Rebase to kubernetes 1.4-pre.rc1

smarterclayton opened the pull request:
* Dropped 1.0 backwards API compatibility (portalIP, node.name)
* Removed the "wait for other backend storage to catch up to resource version" for now with the switch to the new etcd client and the complexity in the storage decorators.  Made SAToken / OAuth Access/Authorize quorum storages (which is significantly faster in etcd3, basically free).  Now means that if 2 masters go down remaining clients can't authorize with service account (so have to fix that).

TODO:

* [ ] Add event recorders to controllers
* [x] Double check nodeName setting in Kubelet and related commands
* [x] Verify etcd paths are correct after changes to REST initialization flows
* [x] Identify why `openshift kubectl` no longer respects `--loglevel` (initialization order?) - could be a result of logs being in a separate package now and being initialized after other code
  * Was a result of cloudflare/cfssl which sets a "loglevel" flag
* [x] Open issues to track new controllers
  * Garbage collection
  * ScheduledJobs
  * ???
* [ ] Work with Jordan on the quorum writes
* [ ] Resolve the controller refs decision

Review TODOs
* [ ] followup for testing that objectmeta field selectors work on all objects
* [ ] https://trello.com/c/DDxyzFL8/60-authorization-collapse-onto-kube-authorization-interface
* [ ] followup issue for `pkg/cmd/cli/cmd/rsync/forwarder.go` ready channel
* [ ] `addKeyToSecret` validation call.
* [ ] followup issue for `func (plugin *OsdnNode) Init(host knetwork.Host` last int arg.
* [ ] follow-up issue to make the ownerref admission plugin check delete permissions and re-enable it
* [ ] followup to ensure that petsets, poddisruption, and scheduledjob controllers default the correct way and at least a scan to make sure they aren't obviously escalating.
* [ ] `ImagePolicyWebhook` admission plugin should be added to the list. disable by default.

smarterclayton commented on pkg/build/api/v1/zz_generated.deepcopy.go:
```diff
@@ -0,0 +1,1079 @@
+			if newVal, err := c.DeepCopy(*in); err != nil {
+				return err
+			} else {
+				*out = newVal.(SourceRevision)
```
@sttts can you help me track down what introduced this regression in deep copy?

```
pkg/build/api/v1/zz_generated.deepcopy.go:339: cannot use newVal.(SourceRevision) (type SourceRevision) as type *SourceRevision in assignment
```

sttts replied:
Looks like you tracked it down yourself yesterday.

You had hit a broken code path in the generator (
https://github.com/kubernetes/kubernetes/blob/bd7d7cc02a370180c25b20096b450b79ba2713b8/cmd/libs/go2idl/deepcopy-gen/generators/deepcopy.go#L612-L612): it must look like this:

```go
sw.Do("*out = newVal.($.|raw$)\n", t)
```

Will fix that upstream.

smarterclayton commented:
Compiles, master starts.

smarterclayton commented:
[test]

liggitt commented on pkg/cmd/server/start/admission.go:
```diff
@@ -79,6 +78,7 @@ var (
 		imagepolicy.PluginName,
 		"BuildOverrides",
 		"AlwaysPullImages",
+		"openshift.io/OwnerReference",
```
follow-up issue to make the admission plugin check delete permissions and re-enable it

smarterclayton commented:
I am not proud of the last commit on this PR

liggitt commented on hack/common.sh:
```diff
@@ -788,6 +793,87 @@ function os::build::ldflags() {
+# os::build::enable_swap attempts to enable swap for the system if a) this is Linux and b)
+# the amount of physical memory is less than 10GB. This is a stopgap until we have
+# better control over memory use in Go 1.7+.
+function os::build::enable_swap() {
```
I would happily contribute my memory if it meant a) we didn't need to do this or b) the thought of it wouldn't haunt me

deads2k replied:
> I would happily contribute my memory if it meant a) we didn't need to do this or b) the thought of it wouldn't haunt me

yikes

sttts replied:
:scream:

marun commented:
@smarterclayton Networking test failure looks to be related to test setup.  A manually deployed dind cluster doesn't exhibit the problem, only one deployed by the test runner.

smarterclayton commented:
Ok, thanks

smarterclayton commented:
@derekwaynecarr  integration test TestExtensionsAPIDisabledAutoscaleBatchEnabled I think is too clever for your namespace cleanup code - it seems to reliably (mostly) trigger a panic in the namespace controller.  I suspect it's because the extension group is completely disabled.  The panic happens after content cleanup appears to succeed.  This is 1.4.0-beta as of two days ago.

```
E0907 22:04:06.356870   78927 runtime.go:64] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/runtime/runtime.go:70
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/runtime/runtime.go:63
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/runtime/runtime.go:49
/usr/local/Cellar/go/1.7/libexec/src/runtime/asm_amd64.s:479
/usr/local/Cellar/go/1.7/libexec/src/runtime/panic.go:458
/usr/local/Cellar/go/1.7/libexec/src/runtime/panic.go:62
/usr/local/Cellar/go/1.7/libexec/src/runtime/sigpanic_unix.go:24
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller_utils.go:380
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller.go:176
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller.go:132
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller.go:152
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller.go:184
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/wait/wait.go:88
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/wait/wait.go:89
/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/wait/wait.go:49
/usr/local/Cellar/go/1.7/libexec/src/runtime/asm_amd64.s:2086
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
       	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x1609f17]

goroutine 7804 [running]:
panic(0x3375020, 0xc4200160f0)
       	/usr/local/Cellar/go/1.7/libexec/src/runtime/panic.go:500 +0x1a1
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
       	/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/runtime/runtime.go:56 +0x126
panic(0x3375020, 0xc4200160f0)
       	/usr/local/Cellar/go/1.7/libexec/src/runtime/panic.go:458 +0x243
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace.syncNamespace(0x5aa6b00, 0xc42a5db7c0, 0x5a64620, 0xc42aef4660, 0xc42e066240, 0xc42df3b800, 0x10, 0x10, 0xc42c8deb40, 0x3a0a0de, ...)
       	/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller_utils.go:380 +0x367
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace.(*NamespaceController).syncNamespaceFromKey(0xc42e018a80, 0xc42e0c91c0, 0x1f, 0x0, 0x0)
       	/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller.go:176 +0x2f3
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace.(*NamespaceController).worker.func1(0x54c00)
       	/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller.go:132 +0x123
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace.(*NamespaceController).worker(0xc42e018a80)
       	/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller.go:152 +0x48
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace.(*NamespaceController).(github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace.worker)-fm()
       	/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller.go:184 +0x2a
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/wait.JitterUntil.func1(0xc42de00b90)
       	/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/wait/wait.go:88 +0x5e
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/wait.JitterUntil(0xc42de00b90, 0x3b9aca00, 0x0, 0x3e42001, 0xc4201c00c0)
       	/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/wait/wait.go:89 +0xad
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/wait.Until(0xc42de00b90, 0x3b9aca00, 0xc4201c00c0)
       	/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/wait/wait.go:49 +0x4d
created by github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace.(*NamespaceController).Run
       	/Users/clayton/projects/origin/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/controller/namespace/namespace_controller.go:184 +0x110
```

smarterclayton commented:
This could actually be the deltafifo - if we got Replace'd and got the version of namespace without TerminationTimestamp set.

smarterclayton commented:
Well that's cute:

```
panic: Invalid state transition: ADDED -> ADDED [recovered]
	panic: Invalid state transition: ADDED -> ADDED
goroutine 94 [running]:
panic(0x44dac40, 0xc8205b59d0)
	/usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/util/runtime/runtime.go:56 +0x153
panic(0x44dac40, 0xc8205b59d0)
	/usr/local/go/src/runtime/panic.go:443 +0x4e9
github.com/openshift/origin/pkg/client/cache.(*EventQueue).handleEvent(0xc8200db7c0, 0x5cec9c0, 0xc82063e5a0, 0x5defbf0, 0x5, 0x0, 0x0)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/pkg/client/cache/eventqueue.go:128 +0xbbb
github.com/openshift/origin/pkg/client/cache.(*EventQueue).Add(0xc8200db7c0, 0x5cec9c0, 0xc82063e5a0, 0x0, 0x0)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/pkg/client/cache/eventqueue.go:192 +0x56
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/client/cache.(*Reflector).watchHandler(0xc8205af380, 0x7fd64388d428, 0xc8204d8810, 0xc820bbbc08, 0xc8205b8b40, 0xc82005c120, 0x0, 0x0)
	/go/src/github.com/openshift/origin/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/client/cache/reflector.go:381 +0xd6b
github.com/openshift/origin/vendor/k8s.io/kubernetes/pkg/client/cache.(*Reflector).ListAndWatch(0xc8205af380, 0xc82005c120, 0x0, 0x0)
```

smarterclayton commented:
[test] copr failure

smarterclayton commented:
[test]

smarterclayton commented:
[test]

smarterclayton commented:
[test]

smarterclayton commented:
[test]

smarterclayton commented:
[test]

smarterclayton commented:
@deads2k looks like taking the lock is failing on the CI machines:

```
https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/5908/testReport/junit/github.com_openshift_origin_vendor_k8s/io_kubernetes_pkg_kubectl_cmd_config/TestUnsetField/
```

smarterclayton commented:
I'm going to delete the whole lock file concept.  It's stupid.

smarterclayton commented:
[test]

smarterclayton commented:
[test]

deads2k commented:
@sttts ptal.

deads2k commented:
Package moves before the bump are all fine and boring.  I'd bet on the commit verifier disliking them.

deads2k commented on pkg/build/api/validation/validation.go:
```diff
@@ -626,7 +626,7 @@ func ValidateBuildLogOptions(opts *buildapi.BuildLogOptions) field.ErrorList {
 }
 
-const cIdentifierErrorMsg string = `must be a C identifier (matching regex ` + kvalidation.CIdentifierFmt + `): e.g. "my_name" or "MyName"`
+var cIdentifierErrorMsg string = `must be a C identifier (mixed-case letters, numbers, and underscores): e.g. "my_name" or "MyName"`
```
Why is this a `var`?

smarterclayton replied:
You can't create a const from a non-const

deads2k commented on pkg/cmd/cli/cmd/rsync/forwarder.go:
```diff
@@ -38,7 +38,8 @@ func (f *portForwarder) ForwardPorts(ports []string, stopChan <-chan struct{}) e
 	}
 	// TODO: Make os.Stdout/Stderr configurable
-	fw, err := portforward.New(dialer, ports, stopChan, f.Out, f.ErrOut)
+	// TODO: Accept a ready channel?
```
followup issue.

smarterclayton replied:
https://github.com/openshift/origin/issues/10908

deads2k commented on pkg/cmd/cli/secrets/new.go:
```diff
@@ -261,8 +261,8 @@ func (o *CreateSecretOptions) BundleSecret() (*kapi.Secret, error) {
 func addKeyToSecret(keyName, filePath string, secretData map[string][]byte) error {
-	if !kvalidation.IsSecretKey(keyName) {
-		return fmt.Errorf("%v is not a valid key name for a secret", keyName)
+	if errors := kvalidation.IsConfigMapKey(keyName); len(errors) > 0 {
```
wrong validation.

smarterclayton replied:
IsSecretKey was removed.

deads2k commented:
Note to self: come back to `Accept cache.TriggerPublisher on ApplyOptions, use StorageFactory`

deads2k commented on pkg/quota/admission/clusterresourcequota/accessor.go:
```diff
@@ -147,10 +148,14 @@ func (e *clusterQuotaAccessor) waitForReadyClusterQuotaNames(namespaceName strin
 		var namespaceSelectionFields clusterquotamapping.SelectionFields
 		clusterQuotaNames, namespaceSelectionFields = e.clusterQuotaMapper.GetClusterQuotasFor(namespaceName)
-		namespace, err := e.namespaceLister.Get(namespaceName)
+		obj, ok, err := e.namespaceLister.Get(&kapi.Namespace{ObjectMeta: kapi.ObjectMeta{Name: namespaceName}})
```
This changed semantics.  Restore the old behavior (fail on namespace get failure) and we'll try to fix the real problem in https://trello.com/c/Issi3JwT/11-controller-standardize-listers-they-should-have-consistent-label-field-selector-behavior-and-should-report-normal-api-errors-loo

smarterclayton replied:
You can't call a Get() with a string.  The semantics are wrong?

deads2k replied:
> You can't call a Get() with a string. The semantics are wrong?

The kube code is wrong and you're handling the `!ok` incorrectly.  As I recall `!ok` is actually a "NotFoundError"

smarterclayton replied:
Ah, thought you were referring to the .Get() call pattern.

deads2k commented on pkg/cmd/server/kubernetes/master_config.go:
```diff
@@ -106,7 +93,95 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
+	}
+
+	resourceEncodingConfig := genericapiserver.NewDefaultResourceEncodingConfig()
+	resourceEncodingConfig.SetVersionEncoding(
```
Where did this list come from? This is going to be a maintenance problem.

smarterclayton replied:
From our old code - this is our encoded historical choices about things we took from kube.

deads2k commented on pkg/cmd/server/kubernetes/master_config.go:
```diff
@@ -106,7 +93,95 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
+		genericapiserver.NewDefaultResourceEncodingConfig(),
+		storageGroupsToEncodingVersion,
+		// FIXME: this GroupVersionResource override should be configurable
+		[]unversioned.GroupVersionResource{batch.Resource("scheduledjobs").WithVersion("v2alpha1")},
```
The duplication here needs refactoring for 1.5.  @soltysh is changing this path for 1.5.  We'll need to be sure we reflect changes here.  issue maybe?

smarterclayton replied:
Yes, will leave this with the general rest storage restgetter issue.

smarterclayton replied:
https://github.com/openshift/origin/issues/10910

deads2k commented on pkg/cmd/server/kubernetes/master_config.go:
```diff
@@ -106,7 +93,95 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
+	}
+
+	kubeletClientConfig := configapi.GetKubeletClientConfig(options)
+	kubeletClient, err := kubeletclient.NewStaticKubeletClient(kubeletClientConfig)
```
I'm not seeing the usage.

deads2k replied:
> I'm not seeing the usage.

nm, found it.

deads2k commented on pkg/sdn/plugin/controller.go:
```diff
@@ -322,6 +322,8 @@ func (plugin *OsdnNode) SetupSDN(localSubnetCIDR, clusterNetworkCIDR, servicesNe
 	itx.IgnoreError()
 	_ = itx.EndTransaction()
 
+	sysctl := sysctl.New()
```
I don't know what this does.  Just make sure its intentional.

smarterclayton replied:
Sysctl now has a method to return a sysctl interface

deads2k commented on pkg/sdn/plugin/plugin.go:
```diff
@@ -53,7 +53,7 @@ func (plugin *OsdnNode) getExecutable() string {
 }
 
-func (plugin *OsdnNode) Init(host knetwork.Host, _ componentconfig.HairpinMode, _ string) error {
+func (plugin *OsdnNode) Init(host knetwork.Host, _ componentconfig.HairpinMode, _ string, _ int) error {
```
This needs a followup issue to figure out what the "right" thing to do is.

smarterclayton replied:
Init of this is done elsewhere - the old interface is going away in 1.5

deads2k commented on pkg/cmd/server/kubernetes/master.go:
```diff
@@ -158,22 +158,24 @@ func (c *MasterConfig) RunNamespaceController(kubeClient internalclientset.Inter
+	alphaProvisioner, err := kctrlmgr.NewAlphaVolumeProvisioner(c.CloudProvider, s.VolumeConfiguration)
 	if err != nil {
-		glog.Fatal("A Provisioner could not be created, but one was expected. Provisioning will not work. This functionality is considered an early Alpha version.")
+		glog.Fatalf("An backward-compatible provisioner could not be created: %v, but one was expected. Provisioning will not work. This functionality is considered an early Alpha version.", err)
```
An backward

deads2k commented on pkg/cmd/server/kubernetes/node.go:
```diff
@@ -280,39 +283,43 @@ func (c *NodeConfig) RunServiceStores(enableProxy, enableDNS bool) {
 func (c *NodeConfig) RunKubelet() {
-	if c.KubeletConfig.ClusterDNS == nil {
+	var clusterDNS net.IP
+	if c.KubeletServer.ClusterDNS == "" {
```
hahahahaha.  Really?  `len(c.KubeletServer.ClusterDNS)`.  That's funny from you.

smarterclayton replied:
I get tired sometimes.

On Sep 14, 2016, at 1:06 PM, David Eads <notifications@github.com> wrote:

In pkg/cmd/server/kubernetes/node.go
<https://github.com/openshift/origin/pull/10777#discussion_r78791071>:

> @@ -280,39 +283,43 @@ func (c *NodeConfig) RunServiceStores(enableProxy, enableDNS bool) {
>
>  // RunKubelet starts the Kubelet.
>  func (c *NodeConfig) RunKubelet() {
> -	if c.KubeletConfig.ClusterDNS == nil {
> +	var clusterDNS net.IP
> +	if c.KubeletServer.ClusterDNS == "" {

hahahahaha. Really? len(c.KubeletServer.ClusterDNS). That's funny from you.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/10777/files/ba9a06d1d23905e3c92cd9aeb98c56290c321f0d..a5c6b78ddd75c9f54dedcc94bd1296c0ac25359b#r78791071>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABG_p50gCNyi_XpVSx0t-EQV-xjvW8uPks5qqCmIgaJpZM4JzJrj>
.

deads2k commented on pkg/cmd/server/kubernetes/node_config.go:
```diff
@@ -232,19 +223,15 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enable
-
-	// Make sure the node doesn't think it is in standalone mode
-	// This is required for the node to enforce nodeSelectors on pods, to set hostIP on pod status updates, etc
-	cfg.StandaloneMode = false
```
@smarterclayton confirm that you checked and this is no longer required for nodeSelectors to work.

smarterclayton replied:
Moved

deads2k commented on pkg/cmd/server/kubernetes/node_config.go:
```diff
@@ -259,11 +246,11 @@ func BuildKubernetesNodeConfig(options configapi.NodeConfig, enableProxy, enable
 
-	if server.CloudProvider == kubeletoptions.AutoDetectCloudProvider {
-		cfg.AutoDetectCloudProvider = true
+	/*if server.CloudProvider == v1alpha1.AutoDetectCloudProvider {
```
This isn't obvious to me.  Why did we just comment this whole section out?

smarterclayton replied:
Already exists inside the Kube code, I need to remove this.

deads2k commented on pkg/cmd/server/origin/rest/storage_options.go:
```diff
@@ -0,0 +1,41 @@
+		options,
+		&genericapiserver.ResourceConfig{},
+		map[unversioned.GroupResource]string{
+			{Resource: "clusterpolicies"}:       "authorization/cluster/policies",
```
issue to inspect this when we start adding API groups.  We're going to need these to agree.

smarterclayton replied:
They provide defaulting, really just a code move from deep in the guts
to one place.  I'll create an issue anyway.

deads2k commented on pkg/config/cmd/cmd.go:
```diff
@@ -165,6 +165,8 @@ func (b *BulkAction) BindForAction(flags *pflag.FlagSet) {
 func (b *BulkAction) BindForOutput(flags *pflag.FlagSet) {
 	flags.StringVarP(&b.Output, "output", "o", "", "Output results as yaml or json instead of executing, or use name for succint output (resource/name).")
 	flags.BoolVar(&b.DryRun, "dry-run", false, "If true, show the result of the operation without performing it.")
+	flags.Bool("no-headers", false, "Omit table headers for default output.")
```
blech.  @fabianofranz this should be fixed upstream.

smarterclayton replied:
PrintOptions needs to be a Struct that we call Bind on that is selective.

On Sep 14, 2016, at 1:20 PM, David Eads <notifications@github.com> wrote:

In pkg/config/cmd/cmd.go
<https://github.com/openshift/origin/pull/10777#discussion_r78793612>:

> @@ -165,6 +165,8 @@ func (b *BulkAction) BindForAction(flags *pflag.FlagSet) {
>  func (b *BulkAction) BindForOutput(flags *pflag.FlagSet) {
>  	flags.StringVarP(&b.Output, "output", "o", "", "Output results as yaml or json instead of executing, or use name for succint output (resource/name).")
>  	flags.BoolVar(&b.DryRun, "dry-run", false, "If true, show the result of the operation without performing it.")
> +	flags.Bool("no-headers", false, "Omit table headers for default output.")

blech. @fabianofranz <https://github.com/fabianofranz> this should be fixed
upstream.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/10777/files/63fd2c03f83157f74a8aca4046256be10c8b7dff..aed693d630231919dee7b15e5a0b0e7e07c182e2#r78793612>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABG_pxBKrEooWGsFOnlb2t14tmZG3dOxks5qqCzsgaJpZM4JzJrj>
.

deads2k commented:
`Remove support for spec.portalIP`.  When did we decide that?  I don't object per-se, but we others like SAName, don't we?

liggitt commented:
> Remove support for spec.portalIP. When did we decide that? I don't object per-se, but we others like SAName, don't we?

We don't carry that, that's upstream

deads2k commented on pkg/cmd/server/start/admission_sync_test.go:
```diff
@@ -37,7 +37,11 @@ var admissionPluginsNotUsedByKube = sets.NewString(
-	"PersistentVolumeLabel", // do we want this? disable by default
+	// "PersistentVolumeLabel", // do we want this? disable by default
+
+	"ImagePolicyWebhook", // from kube, not needed yet
```
I'd rather see this added and disable by default.

smarterclayton replied:
Ok

deads2k commented on pkg/cmd/cli/secrets/new_test.go:
```diff
@@ -60,7 +60,6 @@ func TestCreateSecret(t *testing.T) {
 		{
 			testName: "invalidDNS",
 			args:     []string{"testSecret", "./bsFixtures/invalid/invalid-DNS"},
-			expErr:   true, // "/bsFixtures/invalid-DNS cannot be used as a key in a secret"
```
find another name that isn't allowed.

liggitt replied:
secret volumes can contain multiple directories now?

deads2k commented on pkg/image/admission/imagestream_limits.go:
```diff
@@ -17,79 +12,84 @@ type LimitVerifier interface {
-	}
-	indexer, reflector := cache.NewNamespaceKeyedIndexerAndReflector(lw, &kapi.LimitRange{}, 0)
-	reflector.Run()
+type UninitializedLimitVerifier interface {
```
Where is this used?

smarterclayton replied:
Dead code now, will remove

On Sep 14, 2016, at 1:43 PM, David Eads <notifications@github.com> wrote:

In pkg/image/admission/imagestream_limits.go
<https://github.com/openshift/origin/pull/10777#discussion_r78798076>:

> @@ -17,79 +12,84 @@ type LimitVerifier interface {
>  	VerifyLimits(namespace string, is *imageapi.ImageStream) error
>  }
>
> -func NewLimitVerifier(client kclient.LimitRangesNamespacer) LimitVerifier {
> -	lw := &cache.ListWatch{
> -		ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
> -			return client.LimitRanges(kapi.NamespaceAll).List(options)
> -		},
> -		WatchFunc: func(options kapi.ListOptions) (watch.Interface, error) {
> -			return client.LimitRanges(kapi.NamespaceAll).Watch(options)
> -		},
> -	}
> -	indexer, reflector := cache.NewNamespaceKeyedIndexerAndReflector(lw, &kapi.LimitRange{}, 0)
> -	reflector.Run()
> +type UninitializedLimitVerifier interface {

Where is this used?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/10777/files/12470c9c4ca8ef7c2e76b882442b035f3e1471dd..339b33209a5566822f43f5db3900287735e055f7#r78798076>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABG_p3h1IGBA2_2LCDQ6H_VqGkYrAwzcks5qqDJDgaJpZM4JzJrj>
.

deads2k commented on pkg/image/registry/imagestream/strategy.go:
```diff
@@ -53,9 +54,31 @@ func (s Strategy) NamespaceScoped() bool {
 // and verifies the current user is authorized to access any image streams newly referenced
 // in spec.tags.
+// TODO: this should be part of PrepareForCreate by allowing it to return errors.
+func (s Strategy) BeforeCreate(ctx kapi.Context, obj runtime.Object) error {
```
@mfojtik @miminar this is worth you looking over.

deads2k commented on pkg/deploy/cmd/delete_test.go:
```diff
@@ -120,41 +120,36 @@ func TestStop(t *testing.T) {
 				ktestclient.NewGetAction("replicationcontrollers", "", "config-2"),
 				ktestclient.NewUpdateAction("replicationcontrollers", "", nil),
 				ktestclient.NewGetAction("replicationcontrollers", "", "config-2"),
-				ktestclient.NewGetAction("replicationcontrollers", "", "config-5"),
```
"carry the three, plus two more...", yep this was the line I was looking for.

liggitt replied:
config-5 was my favorite prime config that is the sum of two consecutive primes

smarterclayton replied:
Don't get me started on this test.

On Sep 14, 2016, at 1:49 PM, David Eads <notifications@github.com> wrote:

In pkg/deploy/cmd/delete_test.go
<https://github.com/openshift/origin/pull/10777#discussion_r78799409>:

>  				ktestclient.NewDeleteAction("replicationcontrollers", "", "config-1"),
>  				ktestclient.NewGetAction("replicationcontrollers", "", "config-2"),
>  				ktestclient.NewListAction("replicationcontrollers", "", kapi.ListOptions{}),
>  				ktestclient.NewGetAction("replicationcontrollers", "", "config-2"),
>  				ktestclient.NewUpdateAction("replicationcontrollers", "", nil),
>  				ktestclient.NewGetAction("replicationcontrollers", "", "config-2"),
> -				ktestclient.NewGetAction("replicationcontrollers", "", "config-5"),

"carry the three, plus two more...", yep this was the line I was looking
for.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/10777/files/acd8aac3f65c5e5c1b3c50615fb713841440a693..2751f2c7bdfe639d4596ebf976b546c746918b79#r78799409>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABG_p6Ncis5CRu-l0ex-CxUSXgfZbaIHks5qqDOggaJpZM4JzJrj>
.

deads2k commented on pkg/deploy/cmd/delete_test.go:
```diff
@@ -172,41 +167,36 @@ func TestStop(t *testing.T) {
 				ktestclient.NewGetAction("replicationcontrollers", "", "config-2"),
 				ktestclient.NewUpdateAction("replicationcontrollers", "", nil),
 				ktestclient.NewGetAction("replicationcontrollers", "", "config-2"),
-				ktestclient.NewGetAction("replicationcontrollers", "", "config-5"),
```
oh, but this one down here... I don't know, I liked him.

sttts commented on hack/util.sh:
```diff
@@ -768,10 +768,10 @@ function os::util::host_platform() {
-  if [[ "$(go env GOHOSTOS)" == "darwin" ]]; then
-  	sed -i '' "$@"
-  else
+  if sed --help 2>&1 | grep -qs "GNU sed"; then
```
Is LANG=C set in that context?

Otherwise, the grep fail (with German).

smarterclayton replied:
should lang be on sed or grep?

sttts replied:
On sed. In the German `--help` text they write `GNU-sed`.

sttts commented on pkg/cmd/server/kubernetes/node_config_test.go:
```diff
@@ -135,7 +140,7 @@ func TestProxyConfig(t *testing.T) {
 			IPTablesMasqueradeBit:          &ipTablesMasqueratebit,
 			UDPIdleTimeout:                 unversioned.Duration{Duration: 250 * time.Millisecond},
-			ConntrackMax:                   256 * 1024,                                          // 4x default (64k)
+			ConntrackMaxPerCore:            32 * 1024,                                           // 4x default (64k)
```
wrong comment

sttts replied:
4*64=256

sttts commented on pkg/controller/shared/legacy_kube_informers.go:
```diff
@@ -337,3 +339,115 @@ func (f *namespaceInformer) Lister() *cache.IndexerToNamespaceLister {
+}
+
+func (s storeLimitRangesNamespacer) Get(name string) (*kapi.LimitRange, error) {
+	obj, exists, err := s.indexer.GetByKey(s.namespace + "/" + name)
```
`s.namespace + "/"` is also correct for `s.namespace==""`?

liggitt replied:
limit ranges are namespaced, so every extant limitrange would be stored by a "ns/name" key

deads2k commented:
> deads2k approved these changes just now

Well, that's unfortunate.  Approve a commit and it marks the entire pull.

liggitt commented:
too late. approved. merge.

deads2k commented on contrib/prometheus/prometheus.yml:
```diff
@@ -0,0 +1,49 @@
+  # This is the default relative configuration when you run 'openshift start' from the
+  # root of the source directory.
+  tls_config:
+    ca_file: /var/folders/zw/j6656mbn1rv4fnqqdggss8h80000gn/T//openshift/test-cmd/openshift.local.config/master/ca.crt
```
I don't know what this is, but this doesn't look promising.

deads2k commented on test/cmd/basicresources.sh:
```diff
@@ -129,7 +129,8 @@ os::test::junit::declare_suite_start "cmd/basicresources/services"
 os::cmd::expect_success 'oc delete services frontend'
-os::cmd::expect_failure_and_text 'oc create -f test/integration/testdata/test-service-with-finalizer.json' "finalizers are disabled"
+# TODO: reenable with a permission check
+# os::cmd::expect_failure_and_text 'oc create -f test/integration/testdata/test-service-with-finalizer.json' "finalizers are disabled"
```
note this in your followup issue.

deads2k commented on test/cmd/builds.sh:
```diff
@@ -65,7 +65,7 @@ os::cmd::expect_failure_and_text 'oc new-build ruby-22-centos7~https://github.co
 
 os::cmd::expect_success "oc new-build -D \$'FROM centos:7' --no-output"
-os::cmd::expect_success_and_text 'oc get bc/centos -o=jsonpath="{.spec.output.to}"' '^<nil>$'
+os::cmd::expect_success_and_text 'oc get bc/centos -o=jsonpath="{.spec.output.to}"' '^nil$'
```
add for future release note

deads2k commented:
This needs some serious squashing and you need to handle all the followup requests, but there's nothing I see that would stop me from merging it.

lgetm

smarterclayton commented:
What, you though they'd get it right the first time?

On Sep 14, 2016, at 3:06 PM, David Eads <notifications@github.com> wrote:

deads2k approved these changes just now

Well, that's unfortunate. Approve a commit and it marks the entire pull.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/10777#issuecomment-247120755>,
or mute
the thread
<https://github.com/notifications/unsubscribe-auth/ABG_p_0TbH0e98DOyTNJH1KTFxLxREknks5qqEWrgaJpZM4JzJrj>
.

smarterclayton commented:
Removed both 1.0 API carries we have.  It's been three versions - it's
time David - it's time.

smarterclayton commented:
I'm going to go back and probably make tests run local with constant
paths.  Going to delete this section for now.

On Sep 14, 2016, at 3:08 PM, David Eads <notifications@github.com> wrote:

*@deads2k* commented on this pull request.
------------------------------

In contrib/prometheus/prometheus.yml
<https://github.com/openshift/origin/pull/10777#pullrequestreview-25970>:

> +    labels:
+
+- job_name: openshift_test_cmd
+  scrape_interval:     5s
+
+  honor_labels: true
+  # scrape_interval is defined by the configured global (15s).
+  # scrape_timeout is defined by the global default (10s).
+
+  # metrics_path defaults to '/metrics'
+  # scheme defaults to 'http'.
+
+  # This is the default relative configuration when you run
'openshift start' from the
+  # root of the source directory.
+  tls_config:
+    ca_file: /var/folders/zw/j6656mbn1rv4fnqqdggss8h80000gn/T//openshift/test-cmd/openshift.local.config/master/ca.crt

I don't know what this is, but this doesn't look promising.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/10777#pullrequestreview-25970>,
or mute the thread
<https://github.com/notifications/unsubscribe-auth/ABG_pw4FNzMV2d5vXJbhGou_BERPAs4Xks5qqEYcgaJpZM4JzJrj>
.

miminar commented on pkg/controller/shared/legacy_kube_informers.go:
```diff
@@ -337,3 +339,115 @@ func (f *namespaceInformer) Lister() *cache.IndexerToNamespaceLister {
+	return StoreToLimitRangeLister{Indexer: informer.GetIndexer()}
+}
+
+// StoreToReplicationControllerLister gives a store List and Exists methods. The store must contain only LimitRanges.
```
`/StoreToReplicationControllerLister/StoreToLimitRangeLister/`; Also the `List` and `Exists` method don't seem to exist.

kargakis replied:
The methods are fine since the resource is namespaced (see `storeLimitRangesNamespacer`). What is not fine is the place this code should live (`pkg/client/cache` instead of the shared informer library).

kargakis replied:
Also it needs a TODO to be moved upstream

kargakis commented on pkg/controller/shared/shared_informer.go:
```diff
@@ -35,6 +36,8 @@ type InformerFactory interface {
 	SecurityContextConstraints() SecurityContextConstraintsInformer
 	ClusterResourceQuotas() ClusterResourceQuotaInformer
+
+	KubernetesInformers() informers.SharedInformerFactory
```
This is superseding all the upstream informers that are already exposed by our factory.

kargakis commented on pkg/controller/shared/shared_informer.go:
```diff
@@ -157,3 +160,23 @@ func (f *sharedInformerFactory) SecurityContextConstraints() SecurityContextCons
+}
+func (f kubernetesSharedInformer) PersistentVolumeClaims() informers.PVCInformer {
+	return f.f.PersistentVolumeClaims()
+}
```
You are missing `ReplicationControllers`.

miminar commented on pkg/image/admission/imagestream_limits.go:
```diff
@@ -17,79 +12,79 @@ type LimitVerifier interface {
+	return res, nil
+}
+
+// getMaxLimits finds the limiter provided by the items
```
Outdated godoc.

miminar commented on pkg/image/registry/imagestream/strategy.go:
```diff
@@ -53,9 +54,31 @@ func (s Strategy) NamespaceScoped() bool {
 }
 
-// PrepareForCreate clears fields that are not allowed to be set by end users on creation,
+// BeforeCreate checks a number of creation preconditions before PrepareCreate or Validate is called.
```
`BeforeCreate` is called after `PrepareForCreate`

miminar commented on pkg/image/registry/imagestream/strategy.go:
```diff
@@ -481,33 +487,39 @@ func (s Strategy) prepareForUpdate(obj, old runtime.Object, resetStatus bool) {
-
-// ValidateUpdate is the default update validation for an end user.
-func (s Strategy) ValidateUpdate(ctx kapi.Context, obj, old runtime.Object) field.ErrorList {
+// AdditionalUpdateTransform handles any transformations required before validation or update.
```
`s/AdditionalUpdateTransform/BeforeUpdate/`

kargakis commented on pkg/cmd/server/kubernetes/master.go:
```diff
@@ -264,6 +270,7 @@ func (c *MasterConfig) RunReplicationController(client *client.Client) {
 		kctrlmgr.ResyncPeriod(c.ControllerManager),
 		replicationcontroller.BurstReplicas,
 		int(c.ControllerManager.LookupCacheSizeForRC),
+		c.ControllerManager.EnableGarbageCollector,
```
Is there an issue for switching all controllers to use the garbage collector?

soltysh commented:
> The duplication here needs refactoring for 1.5. @soltysh is changing this path for 1.5. We'll need to be sure we reflect changes here. issue maybe?

@deads2k aside from the changes I've also filled in an issue to fix the manual enablement for `batch/v2alpha1` https://github.com/kubernetes/kubernetes/issues/32661

soltysh commented on pkg/cmd/server/kubernetes/master_config.go:
```diff
@@ -106,7 +93,95 @@ func BuildKubernetesMasterConfig(options configapi.MasterConfig, requestContextM
+		master.DefaultAPIResourceConfigSource(),
+	)*/
+	// the order here is important, it defines which version will be used for storage
+	storageFactory.AddCohabitatingResources(extensions.Resource("jobs"), batch.Resource("jobs"))
```
This might be problematic soon, since I'm planning to remove Jobs from extensions in k8s 1.5, see https://github.com/kubernetes/kubernetes/issues/32763

kargakis commented on pkg/cmd/cli/kubectl_compat_test.go:
```diff
@@ -26,6 +26,7 @@ var MissingCommands = sets.NewString(
 	"drain",
 	"uncordon",
 	"taint",
+	"top",
```
Add a comment that we use this as part of oadm

kargakis commented on pkg/cmd/server/start/start_master.go:
```diff
@@ -629,28 +638,37 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
 
+		policyEnabled := len(configapi.GetEnabledAPIVersionsForGroup(kc.Options, policy.GroupName)) > 0
+		if policyEnabled {
+			kc.RunDisruptionBudgetController(disruptionClient)
```
Do we really want to run this controller now? There is still some work that needs to happen for it.

soltysh commented on pkg/cmd/server/start/start_master.go:
```diff
@@ -629,28 +638,37 @@ func startControllers(oc *origin.MasterConfig, kc *kubernetes.MasterConfig) erro
+		batchEnabled := len(configapi.GetEnabledAPIVersionsForGroup(kc.Options, batch.GroupName)) > 0
+		if extensionsEnabled || batchEnabled {
 			kc.RunJobController(jobClient)
+			kc.RunScheduledJobController(jobConfig)
```
You can't enable SJ when only extensions are enabled. SJ is only for batch.

soltysh replied:
Even more, upstream specifically check SJ resource existence, since you allow per-resource enablement, which I guess is not applicable to origin, is it?

soltysh replied:
Additionally, how one disables, or rather enables SJ. They are alpha and I'd rather not to run them by default but would prefer them being turned on explicitly through `--alpha` flag or similar.

soltysh commented on test/testdata/bootstrappolicy/bootstrap_cluster_roles.yaml:
```diff
@@ -62,6 +62,7 @@ items:
     - persistentvolumes/status
```
Can you squash this commit along with 'Add scheduled jobs to all roles, new endpoints to cluster-reader role' ?

soltysh replied:
There two more commits 'DaemonSet controller needs to be able to get daemonsets' and 'Allow deployer to watch replication controllers' touching those. How about squashing all of that into one commit?

soltysh commented on pkg/security/admission/admission_test.go:
```diff
@@ -1129,7 +1128,8 @@ func createNamespaceForTest() *kapi.Namespace {
 		ObjectMeta: kapi.ObjectMeta{
-			Name: "default",
+			Name:      "default",
+			Namespace: "default",
```
Use `kapi.NamespaceDefault` instead

mfojtik commented on test/extended/deployments/deployments.go:
```diff
@@ -379,7 +379,7 @@ var _ = g.Describe("deploymentconfigs", func() {
 			o.Expect(err).NotTo(o.HaveOccurred())
 			g.By(fmt.Sprintf("checking the history for substrings\n%s", out))
-			o.Expect(out).To(o.ContainSubstring("deploymentconfigs \"deployment-simple\" history viewed"))
+			o.Expect(out).To(o.ContainSubstring("deploymentconfigs \"deployment-simple\""))
```
@kargakis FYI

kargakis replied:
this is fine, the upstream PR ended up not including it: https://github.com/kubernetes/kubernetes/pull/27267

kargakis replied:
And we should probably drop that header from `rollout history` once and forever

mfojtik commented on pkg/deploy/cmd/delete_test.go:
```diff
@@ -221,13 +211,12 @@ func TestStop(t *testing.T) {
 			},
 			kexpected: []ktestclient.Action{
-				ktestclient.NewListAction("replicationcontrollers", "default", kapi.ListOptions{}),
+				ktestclient.NewListAction("replicationcontrollers", "default", kapi.ListOptions{LabelSelector: labels.SelectorFromSet(map[string]string{"openshift.io/deployment-config.name": "config"})}),
```
`deployapi.DeploymentConfigAnnotation`

liggitt replied:
no, this test should smack us if we change the constant's value and break people

mfojtik commented on test/integration/v2_docker_registry_test.go:
```diff
@@ -119,6 +121,10 @@ middleware:
 
 	go dockerregistry.Execute(strings.NewReader(config))
 
+	if err := cmdutil.WaitForSuccessfulDial(false, "tcp", "127.0.0.1:5000", 100*time.Millisecond, 1*time.Second, 35); err != nil {
```
@miminar PTAL

miminar replied:
LGTM

smarterclayton commented:
We don't bother with that yet.

On Thu, Sep 15, 2016 at 7:05 AM, Maciej Szulik <notifications@github.com>
wrote:

> *@soltysh* commented on this pull request.
> ------------------------------
>
> In pkg/cmd/server/start/start_master.go
> <https://github.com/openshift/origin/pull/10777>:
>
> >  			kc.RunJobController(jobClient)
> +			kc.RunScheduledJobController(jobConfig)
>
> Additionally, how one disables, or rather enables SJ. They are alpha and
> I'd rather not to run them by default but would prefer them being turned on
> explicitly through --alpha flag or similar.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10777>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p8LV-Bcppu6pj0OtNj4Uvdo7u4iLks5qqSZ4gaJpZM4JzJrj>
> .
>

smarterclayton commented:
Follow up issue.  It's not degenerate in its behavior so no impact now.

On Thu, Sep 15, 2016 at 6:57 AM, Michail Kargakis <notifications@github.com>
wrote:

> *@kargakis* commented on this pull request.
> ------------------------------
>
> In pkg/cmd/server/start/start_master.go
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-139281>:
>
> >  			kc.RunHPAController(hpaOClient, hpaKClient, oc.Options.PolicyConfig.OpenShiftInfrastructureNamespace)
>  		}
>  		if extensionsEnabled {
>  			kc.RunDaemonSetsController(daemonSetClient)
>  		}
>
> +		policyEnabled := len(configapi.GetEnabledAPIVersionsForGroup(kc.Options, policy.GroupName)) > 0
> +		if policyEnabled {
> +			kc.RunDisruptionBudgetController(disruptionClient)
>
> Do we really want to run this controller now? There is still some work
> that needs to happen for it.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-139281>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p6mfpoU9eZEj-D0IcUGISZIkXbkWks5qqSSlgaJpZM4JzJrj>
> .
>

smarterclayton commented:
Open a follow up issue for origin.

On Thu, Sep 15, 2016 at 6:45 AM, Maciej Szulik <notifications@github.com>
wrote:

> *@soltysh* commented on this pull request.
> ------------------------------
>
> In pkg/cmd/server/kubernetes/master_config.go
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-137334>:
>
> > +		[]unversioned.GroupVersionResource{batch.Resource("scheduledjobs").WithVersion("v2alpha1")},
> +		master.DefaultAPIResourceConfigSource(), server.RuntimeConfig,
> +	)
> +	if err != nil {
> +		return nil, nil, err
> +	}
> +
> +	/*storageFactory := genericapiserver.NewDefaultStorageFactory(
> +		etcdConfig,
> +		server.DefaultStorageMediaType,
> +		kapi.Codecs,
> +		resourceEncodingConfig,
> +		master.DefaultAPIResourceConfigSource(),
> +	)*/
> +	// the order here is important, it defines which version will be used for storage
> +	storageFactory.AddCohabitatingResources(extensions.Resource("jobs"), batch.Resource("jobs"))
>
> This might be problematic soon, since I'm planning to remove Jobs from
> extensions in k8s 1.5, see kubernetes/kubernetes#32763
> <https://github.com/kubernetes/kubernetes/issues/32763>
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-137334>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_pyATNSSLloKWhXUEexzOXzFKcn54ks5qqSHNgaJpZM4JzJrj>
> .
>

smarterclayton commented:
Already has one, 5 lines up.

On Thu, Sep 15, 2016 at 6:55 AM, Michail Kargakis <notifications@github.com>
wrote:

> *@kargakis* commented on this pull request.
> ------------------------------
>
> In pkg/cmd/cli/kubectl_compat_test.go
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-138979>:
>
> > @@ -26,6 +26,7 @@ var MissingCommands = sets.NewString(
>  	"drain",
>  	"uncordon",
>  	"taint",
> +	"top",
>
> Add a comment that we use this as part of oadm
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-138979>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p9FG4vIPbMUN5S4IseB0yYg58dSrks5qqSQrgaJpZM4JzJrj>
> .
>

smarterclayton commented:
You guys probably need to create a card on your board for reacting to GC.
I think I'm not eager to enable it for 1.4.

On Thu, Sep 15, 2016 at 5:21 AM, Michail Kargakis <notifications@github.com>
wrote:

> *@kargakis* commented on this pull request.
> ------------------------------
>
> In pkg/cmd/server/kubernetes/master.go
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-123791>:
>
> > @@ -264,6 +270,7 @@ func (c *MasterConfig) RunReplicationController(client *client.Client) {
>  		kctrlmgr.ResyncPeriod(c.ControllerManager),
>  		replicationcontroller.BurstReplicas,
>  		int(c.ControllerManager.LookupCacheSizeForRC),
> +		c.ControllerManager.EnableGarbageCollector,
>
> Is there an issue for switching all controllers to use the garbage
> collector?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-123791>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p6YrvFkYXe7w5UtL0OidAOOCURlBks5qqQ4HgaJpZM4JzJrj>
> .
>

smarterclayton commented:
Not according to Go.

On Thu, Sep 15, 2016 at 5:06 AM, Michail Kargakis <notifications@github.com>
wrote:

> *@kargakis* commented on this pull request.
> ------------------------------
>
> In pkg/controller/shared/shared_informer.go
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-121217>:
>
> > +
> +// kubernetesSharedInformer adapts this informer factory to the identical interface as kubernetes
> +type kubernetesSharedInformer struct {
> +	f *sharedInformerFactory
> +}
> +
> +func (f kubernetesSharedInformer) Start(ch <-chan struct{})                { f.f.Start(ch) }
> +func (f kubernetesSharedInformer) Pods() informers.PodInformer             { return f.f.Pods() }
> +func (f kubernetesSharedInformer) Namespaces() informers.NamespaceInformer { return f.f.Namespaces() }
> +func (f kubernetesSharedInformer) Nodes() informers.NodeInformer           { return f.f.Nodes() }
> +func (f kubernetesSharedInformer) PersistentVolumes() informers.PVInformer {
> +	return f.f.PersistentVolumes()
> +}
> +func (f kubernetesSharedInformer) PersistentVolumeClaims() informers.PVCInformer {
> +	return f.f.PersistentVolumeClaims()
> +}
>
> You are missing ReplicationControllers.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-121217>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_p1lLbPbw4VSGqApipIGyaDFMK-f6ks5qqQqSgaJpZM4JzJrj>
> .
>

smarterclayton commented:
No it isn't?

On Thu, Sep 15, 2016 at 5:05 AM, Michail Kargakis <notifications@github.com>
wrote:

> *@kargakis* commented on this pull request.
> ------------------------------
>
> In pkg/controller/shared/shared_informer.go
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-121090>:
>
> > @@ -35,6 +36,8 @@ type InformerFactory interface {
>  	ImageStreams() ImageStreamInformer
>  	SecurityContextConstraints() SecurityContextConstraintsInformer
>  	ClusterResourceQuotas() ClusterResourceQuotaInformer
> +
> +	KubernetesInformers() informers.SharedInformerFactory
>
> This is superseding all the upstream informers that are already exposed by
> our factory.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/10777#pullrequestreview-121090>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_py1g7BKIoYYb_fv-sKnRncJJUdkYks5qqQpfgaJpZM4JzJrj>
> .
>

smarterclayton commented:
[test]

smarterclayton commented:
[test]

smarterclayton commented:
Ginkgo dropped out, we may be taking too long in one code path.  nothing obvious in logs.

smarterclayton commented:
[merge]
